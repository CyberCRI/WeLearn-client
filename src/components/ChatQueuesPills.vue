<script setup lang="ts">
defineProps<{
  action: (message: string) => void;
  messageList: string[];
  shouldDisable: boolean;
}>();
</script>
<template>
  <div class="queues scroll" :class="{ disabled: shouldDisable }">
    <div class="flexy">
      <div
        v-for="message in messageList.slice(-2)"
        :key="message"
        class="card cursor"
        @click="action(message)"
      >
        <div class="card-content">
          {{ message }}
        </div>
      </div>
    </div>
  </div>
</template>
<style scoped>
.cursor {
  cursor: pointer;
}
.queues.flexy {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: var(--neutral-0);
  overflow-x: scroll;
  width: 100%;
  margin: 0.5rem 0rem;
}

.queues.disabled {
  margin: 0;
  height: 1.5em;
  overflow: hidden;
}

.flexy {
  display: flex;
  flex-direction: column;
  align-self: flex-start;
  align-items: flex-start;
  gap: 0.5rem;
}

.queues::-webkit-scrollbar {
  width: 0.3125rem;
  height: 0.25rem;
  background-color: var(--neutral-90);
}

/* Add a thumb */
.queues::-webkit-scrollbar-thumb {
  background: var(--neutral-50);
}

@media screen and (min-width: 768px) {
  .queues {
    display: flex;
    align-items: center;
    justify-content: center;
    color: var(--neutral-0);
    overflow-y: auto;
    overflow-x: hidden;
    scrollbar-color: var(--neutral-50) var(--neutral-90);
    scrollbar-width: thin;
    width: 100vw;
    margin: 0.5rem 0rem;
  }

  .flexy {
    display: flex;
    align-self: center;
    align-items: flex-start;
    flex-direction: row;
    gap: 0.5rem;
    padding: 0.5rem;
    margin: 0 10px 0 10px;
    overflow: hidden;
    height: 100%;
    & > div {
      height: 100%;
      justify-content: center;
      align-items: center;
      margin: 0;
    }
  }
}
</style>
