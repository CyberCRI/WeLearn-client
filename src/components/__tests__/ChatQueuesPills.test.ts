import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import ChatQueuesPills from '../ChatQueuesPills.vue';

describe('CheckboxCompnentVue', () => {
  it('renders properly', () => {
    const wrapper = mount(ChatQueuesPills, {
      propsData: {
        messageList: ['i am a question 1', 'i am a question 2'],
        action: (arg: string) => arg,
        shouldDisable: false
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.text()).toContain('i am a question 1');
    expect(wrapper.findAll('.card')).toHaveLength(2);
  });

  it('should the questions with disabled class', () => {
    const wrapper = mount(ChatQueuesPills, {
      propsData: {
        messageList: ['i am a question 1', 'i am a question 2'],
        action: (arg: string) => arg,
        shouldDisable: true
      }
    });
    expect(wrapper.findAll('.disabled')).toHaveLength(1);
  });
});
