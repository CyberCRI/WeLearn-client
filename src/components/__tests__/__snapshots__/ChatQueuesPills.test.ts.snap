// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CheckboxCompnentVue > renders properly 1`] = `
<div
  class="queues scroll"
  data-v-b972df67=""
>
  <div
    class="flexy"
    data-v-b972df67=""
  >
    
    <div
      class="card cursor"
      data-v-b972df67=""
    >
      <div
        class="card-content"
        data-v-b972df67=""
      >
        i am a question 1
      </div>
    </div>
    <div
      class="card cursor"
      data-v-b972df67=""
    >
      <div
        class="card-content"
        data-v-b972df67=""
      >
        i am a question 2
      </div>
    </div>
    
  </div>
</div>
`;
